hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -2191,4 +2193,22 @@ private V mergeLocally(K key, V value, BiFunction<? super V, ? super V, ? extend
         }
     }
 
+    @Override
+    public void replaceAll(@Nonnull BiFunction<? super K, ? super V, ? extends V> function) {
+        checkNotNull(function, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+        replaceAllInternal(function);
+    }
+
+    protected void replaceAllInternal(BiFunction<? super K, ? super V, ? extends V> function) {
+        if (SerializationUtil.isClassStaticAndSerializable(function)) {
+            int partitionCount = getContext().getPartitionService().getPartitionCount();
+            Data functionAsData = toData(function);
+            for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+                ClientMessage request = MapReplaceAllCodec.encodeRequest(name, functionAsData);
+                new ClientInvocation(getClient(), request, getName(), partitionId).invoke();
```

Comment:
I think we need to wait for the future to finish eventough we are not returning any response. Can you add wait part similar to getAll ?

Reply:
Done